File: apigatewayv2-go-lambda/lambda.go
package main

import (
	"fmt"
	"net/http"

	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-lambda-go/lambda"
)

// The input type and the output type are defined by the API Gateway.
func handleRequest(req events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
	fmt.Printf("Body, %s!\n", req.Body)
	name, ok := req.QueryStringParameters["name"]
	if !ok {
		res := events.APIGatewayProxyResponse{
			StatusCode: http.StatusBadRequest,
		}
		return res, nil
	}

	res := events.APIGatewayProxyResponse{
		StatusCode: http.StatusOK,
		Headers:    map[string]string{"Content-Type": "text/plain; charset=utf-8"},
		Body:       fmt.Sprintf("Hello, %s!\n", name),
	}
	return res, nil
}

func main() {
	lambda.Start(handleRequest)
}

File: apigateway-rest-lambda-api-key/lambda.js
exports.handler = function(event, context, callback) {
		console.log('Received event:', JSON.stringify(event, null, 2));

		console.log(event["headers"]["x-api-key"])
		console.log(event["body"])

		var res ={
				"statusCode": 200,
				"headers": {
						"Content-Type": "*/*"
				}
		};
		callback(null, res);
};
